### PR TITLE
Add point cluster renderer description

### LINKED ISSUE #1434: [FEATURE] Point cluster renderer
Original commit: https://github.com/qgis/QGIS/commit/36e276f993d5620740d23941ab58aea8d549c6b5 by nyalldawson

Groups nearby points into a single rendered marker symbol.

QgsPointDisplacementRenderer has been split into a new
pure virtual QgsPointDistanceRenderer base class which handles the
detection of clusters and grouping of points. The new cluster
renderer reuses this base class to avoid code duplication.

Additionally, some improvements have been made to the
displacement renderer, specifically:
- points are now assigned to the group which is "nearest" them,
  rather then just assigning them first group within the search
  distance. In some cases this was assigning features to a more
  distant cluster, resulting in less predictable cluster patterns
- individual points are now correctly shown in their own
  selection state

Lots of code cleanup + documentation too.

Sponsored by:
- Andreas Neumann
- Qtibia Engineering (Tudor Barascu)
- Karl-Magnus Jönsson
- Geonesia (Nicolas Ponzo)
- Plus numerous additional anonymous backers whose generous
  contributions are also highly valued!

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #1434

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -534,6 +534,37 @@ around one center symbol or on several concentric circles.
...
+
+Unlike the |pointDisplacementSymbol| :guilabel:`Point Displacement` renderer
+which blows up overlaid point features placement, the |pointClusterSymbol|
+:guilabel:`Point Cluster` renderer groups nearby points into a single
+rendered marker symbol. Based on a specified :guilabel:`Distance`, points
+that fall within from each others are merged into a single symbol.
+Points aggregation is made based on the closest group being formed, rather
+than just assigning them the first group within the search distance.
+
+You can change the :guilabel:`cluster symbol` rendering, which by default
```
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Capital letter in `Cluster symbol`

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -534,6 +534,37 @@ around one center symbol or on several concentric circles.
...
+displays the number of aggregated features thanks to the ``@cluster_size``
+variable. A ``@cluster_color`` variable storing the cluster symbol color is
+also generated.
```
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Not sure if it's clear where you can change this.

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -534,6 +534,37 @@ around one center symbol or on several concentric circles.
...
+The :guilabel:`Renderer` combobox from which you can choose any of the common
+feature rendering types helps you symbolize the layer based on feature's
+attributes. Push the **[Renderer settings...]** button to configure it.
+Note that this renderer is the one shown on the point features when they are
+not inside a group.
+
+.. note::
+
+ Point displacement and cluster renderers do not alter feature geometry,
+ meaning that points are not moved from their position. They still are located
```
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: They are still located?

**Comment by DelazJ** [comment]
@SrNetoChan I rewrote the description; i wasn't satisfied by the place where `cluster_color` appeared. still not sure though. If you have any comment....
Thanks

**Pull request merged**